gpytorch/posterior/posterior_strategy.py

```diff
@@ -26,6 +42,34 @@ def exact_posterior_alpha(self, train_mean, train_y):
         """
         raise NotImplementedError
 
+    def exact_posterior_lanczos(self):
+        """
+        Computes the lanczos decomposition of the train_train_covariance matrix
+        ((Lazy)Variable nxn)
+
+        k -- the rank of the Lanczos decomposition -- is determined by
+        gpytorch.functions.max_lanczos_iterations
+
+        Returns:
+            - q_mat (matrix (nxk)) - Lanczos Q matrix of train/train covariance matrix
+            - t_mat (matrix (kxk)) - Lanczos Q matrix of train/train covariance matrix
+        """
+        if isinstance(self.var, LazyVariable):
+            train_train_representation = [var.data for var in self.var.representation()]
+            train_train_matmul = self.var._matmul_closure_factory(*train_train_representation)
+            tensor_cls = type(train_train_representation[0])
+        else:
+            train_train_matmul = self.var.data.matmul
+            tensor_cls = type(self.var.data)
+
+        n_train = self.var.size()[0]
+        max_iter = min(n_train, gpytorch.functions.max_lanczos_iterations)
+        lq_object = StochasticLQ(cls=tensor_cls, max_iter=max_iter)
+        init_vector = tensor_cls(n_train, 1).normal_()
+        init_vector /= torch.norm(init_vector, 2, 0)
+        q_mat, t_mat = lq_object.lanczos_batch(train_train_matmul, init_vector)
```

Comment:
We might want to add the eigenvalue binary search for T.

Reply:
As far as I can tell, this is the same LQ code that we use in mul lazy variable. Since it's not going to be the default option (and hopefully we'll have more stable LQ code soon) should we just merge as is?